parsl/executors/high_throughput/executor.py

```diff
@@ -0,0 +1,397 @@
+                 interchange_port_range=(55000, 56000),
+                 storage_access=None,
+                 working_dir=None,
+                 engine_debug=False,
+                 managed=True):
+
+        logger.debug("Initializing HighThroughputExecutor")
+
+        self.label = label
+        self.launch_cmd = launch_cmd
+        self.provider = provider
+        self.engine_debug = engine_debug
+        self.storage_access = storage_access if storage_access is not None else []
+        if len(self.storage_access) > 1:
+            raise ConfigurationError('Multiple storage access schemes are not supported')
+        self.working_dir = working_dir
+        self.managed = managed
+        self.engines = []
+        self.tasks = {}
+
+        self.public_ip = public_ip
+        self.worker_ports = worker_ports
+        self.worker_port_range = worker_port_range
+        self.interchange_port_range = interchange_port_range
+
+        if not launch_cmd:
+            self.launch_cmd = """process_worker_pool.py {debug} -w {tasks_per_node} --task_url={task_url} --result_url={result_url}"""
+
+    def start(self):
+        """Create the Interchange process and connect to it.
+        """
+        self.outgoing_q = zmq_pipes.TasksOutgoing("127.0.0.1", self.interchange_port_range)
+        self.incoming_q = zmq_pipes.ResultsIncoming('127.0.0.1', self.interchange_port_range)
+
+        self.is_alive = True
+
+        self._queue_management_thread = None
+        self._start_queue_management_thread()
+        self._start_local_queue_process()
+
+        logger.debug("Created management thread : %s", self._queue_management_thread)
+
+        if self.provider:
+            debug_opts = "--debug" if self.engine_debug else ""
+            l_cmd = self.launch_cmd.format(debug=debug_opts,
+                                           task_url=self.worker_task_url,
+                                           result_url=self.worker_result_url,
+                                           tasks_per_node=self.provider.tasks_per_node,
+                                           nodes_per_block=self.provider.nodes_per_block)
+            self.launch_cmd = l_cmd
+            logger.debug("Launch command :{}".format(self.launch_cmd))
+
+            self._scaling_enabled = self.provider.scaling_enabled
+            logger.debug("Starting HighThroughputExecutor with provider:\n%s", self.provider)
+            if hasattr(self.provider, 'init_blocks'):
+                try:
+                    for i in range(self.provider.init_blocks):
+                        engine = self.provider.submit(self.launch_cmd, 1)
+                        logger.debug("Launched block: {0}:{1}".format(i, engine))
+                        if not engine:
+                            raise(ScalingFailed(self.provider.label,
+                                                "Attempts to provision nodes via provider has failed"))
+                        self.engines.extend([engine])
```

Comment:
From this loop, it looks like `self.engines` is actually a list of blocks. Is that correct? Can we clarify the naming?